### ISSUE TITLE
Not all server plans are in PriceZone

### ISSUE THREAD

**Issue by derkgort** [UpCloudLtd/upcloud-php-api#20]
Hi,

Not all server plans are defined in PriceZone.

Only these 2:

        'server_plan_1x_cpu_1_gb' => '\Upcloud\ApiClient\Model\Price',
        'server_plan_2x_cpu_2_gb' => '\Upcloud\ApiClient\Model\Price'

https://github.com/UpCloudLtd/upcloud-php-api/blob/master/lib/Model/PriceZone.php#L63

This is the full list: 

server_plan_1xCPU-1GB
server_plan_1xCPU-2GB
server_plan_20xCPU-128GB
server_plan_20xCPU-96GB
server_plan_2xCPU-4GB
server_plan_4xCPU-8GB
server_plan_6xCPU-16GB
server_plan_8xCPU-32GB

Is there a reason they are not defined?

**Comment by derkgort** [comment]
I just added pull request: #21 
Not sure that's the way it's done, because it looks like generated code.

I also added a function to get the plan based on the value returned in ServerApi.

**Comment by Darep** [comment]
Yeah, I think the plans were generated with swagger from some incomplete yaml file by the original author of this library... IMO the best solution would be to remove all hard-coded plans/string, and use the plans API: GET /1.2/plan https://developers.upcloud.com/7-plans/ - and pass server create etc. the "name" of the plan you want. Just an idea though...

**Comment by alienhaxor** [comment]
Thank you for creating the issue and providing the PR. https://github.com/UpCloudLtd/upcloud-php-api/pull/21

For now all server plans are in PriceZone.

I agree with Darep's solution and since we will face this issue again in the future, I'll leave this issue open for now.

**Comment by samir-haliru** [comment]
Hi derkgort! This issue has been open since 2019 without recent activity. We're cleaning up old issues across our repositories.

The UpCloud API /price response includes pricing for many server_plan_* variants (including newer plan families like cloud-native, DEV, and GPU). The PHP client's PriceZone model only exposes typed getter methods for a small set of older plans, so prices for newer plan keys aren't accessible via the PriceZone getters.

Because new plan keys are added over time, hard-coding every plan into the model will always drift. A sustainable "fix" would be exposing plan prices via a dynamic map/accessor rather than adding more getters.

As a workaround, instead of using the PriceZone getters, treat the /price response as an associative array/map (read server_plan_* keys directly), or use the /plan endpoint.

If you still need a library-side fix, please reply with:

- the version of upcloudltd/upcloud-php-api you're using
- what you'd expect the SDK to provide (e.g. getPlanPrice('server_plan_1xCPU-1GB') or a map of planKey -> Price)
- a short code snippet showing your use case

If we don't hear back in 14 days, we'll close this issue. You can reopen it or create a new one anytime if it's still affecting you.

**Comment by derkgort** [comment]
Hi,

I'm not sure how I solved it, and what I'm using now, but i guess I'm already doing something like that.
It's been so long that you can close this I guess. If I ever need to do anything with this I'll probably take the go route.

Thanks!
